sql-plugin/src/main/scala/com/nvidia/spark/rapids/NvtxRangeWithDoc.scala

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids
+
+import ai.rapids.cudf.{NvtxColor, NvtxRange}
+import java.io.{File, FileOutputStream}
+import scala.collection.mutable
+
+sealed case class NvtxId private(name: String, color: NvtxColor, doc: String) {
+  def help(): Unit = println(s"$name|$doc")
+
+  def apply[V](block: => V): V = {
+    try {
+      NvtxRange.pushRange(name, color)
+      block
+    } finally {
+      NvtxRange.popRange()
+    }
+  }
+}
+
+object NvtxRegistry {
+  val registeredRanges: mutable.Map[String, NvtxId] = mutable.Map[String, NvtxId]()
+
+  private def register(id: NvtxId): Unit = {
+    if (registeredRanges.contains(id.name)) {
+      Console.err.println(s"Collision detected for key: ${id.name}")
```

Comment:
Will this ever be noticed? Why not throw here that it can caught in tests and fixed?

Reply:
I can change it to `throw new IllegalArgumentException(s"Collision detected for key: ${id.name}")`